multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1505,11 +1507,80 @@ def evaluate_coco(
         cocoEval.accumulate()
         cocoEval.summarize()
 
+    def _realtime_predict(
+        self,
+        data: Union[pd.DataFrame, dict, list],
+        requires_label: bool,
+    ) -> List[Dict]:
+        data, df_preprocessor, data_processors = self._on_predict_start(
+            config=self._config,
+            data=data,
+            requires_label=requires_label,
+        )
+
+        lengths = []
+        modality_features = {}
+        for per_modality in data_processors:
+            per_modality_features = getattr(df_preprocessor, f"transform_{per_modality}")(data)
+            modality_features[per_modality] = per_modality_features
+            if per_modality_features:
+                lengths.append(len(per_modality_features[next(iter(per_modality_features))]))
+        assert len(set(lengths)) == 1  # make sure each modality has the same sample num
+        sample_num = lengths[0]
+
+        processed_features = []
+        for i in range(sample_num):
+            per_item_features = {}
+            for per_modality, per_modality_processors in data_processors.items():
+                for per_model_processor in per_modality_processors:
+                    if modality_features[per_modality]:
+                        per_item_features.update(
+                            per_model_processor(modality_features[per_modality], idx=i, is_training=False)
+                        )
+            processed_features.append(per_item_features)
+
+        collate_fn = get_collate_fn(df_preprocessor=df_preprocessor, data_processors=data_processors)
+        batch = collate_fn(processed_features)
```

Comment:
Here is the batch we need for ONNX export/evaluation. We can modularize code above in next pr. For now I'll copy this code into export_onnx.